### PR TITLE
Add documentation on how to enable watch API

### PR DESCRIPTION
**Pull request by josephschorr**

### CODE REVIEW

File: docs/spicedb/enabling-watch-api.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Enabling Watch API
+
+The [Watch API] is by default enabled for most [datastores].
+
+[watch api]: https://buf.build/authzed/api/docs/main:authzed.api.v1#authzed.api.v1.WatchService.Watch
+[datastores]: selecting-a-datastore
+
+However, Postgres and CockroachDB require special flags in order to enable the watch API.
```
Comment (jzelinskie) [inline docs/spicedb/enabling-watch-api.md:8]: I would include why this isn't done for them.

File: docs/spicedb/enabling-watch-api.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Enabling Watch API
```
Comment (jzelinskie) [inline docs/spicedb/enabling-watch-api.md:1]: I'm not sure this should be in its own article, but rather a section in the data stores doc

**Review by jzelinskie — CHANGES_REQUESTED** [review]

File: docs/spicedb/enabling-watch-api.md (hunk 2 above)
Comment (josephschorr) [reply docs/spicedb/enabling-watch-api.md:1]: I want it distinct for easy linking

**Comment by josephschorr** [comment]
Updated

File: docs/spicedb/enabling-watch-api.md (hunk 2 above)
Comment (jzelinskie) [reply docs/spicedb/enabling-watch-api.md:1]: Except it makes it less discoverable. If you are reading the datastore docs you'd have no idea that you'd have to follow instructions on another doc. Maybe if we add links from there to here, that'd be sufficient

Comment (josephschorr) [reply docs/spicedb/enabling-watch-api.md:1]: I'll add links

**Review by jzelinskie — APPROVED** [review]

**Pull request merged**